src/worker/task.js

```diff
@@ -0,0 +1,55 @@
+'use babel'
+
+// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+import { Task } from 'atom'
+
+const createWorkerTask = () => {
+  let task = null
+  let started = false
+  const isConnected = () => !!(task && task.childProcess.connected)
+
+  /**
+   * Start the worker process if it hasn't already been started
+   */
+  const start = () => {
```

Comment:
These functions are being redeclared each time the function is called, why not move them out to the file scope along with the  associated variables?

Reply:
This is creating what is, for all intents, an `instance`. Call `createWorkerTask` and it wraps all these variables in a closure creating state for a worker and returning an object with the api to interact with it. Then if you choose to call `createWorkerTask` again, you get a new closure, with new variables, and a new worker task. If you use only 1 worker, then you only call it once, and it doesn't matter that it's recreating. If we want to grab a backup worker in case of failure, then this (isn't quite there yet but) is moving in the direction of making that easy.

Reply:
I know you hate them, but that's _exactly_ what a `class` is for (and _also_ stops re-declaring the methods on each call). 😛

Reply:
Thanks for the input. I'll leave it like this for now and take that into consideration as I move forward. Mostly it was just a way to isolate these worker calls away from the stuff using it so I could get it into a separate file. It gets that job done, so I can keep looking for other pieces to break out.

At the current stage, I'm trying to minimize changes to existing code beyond breaking it into smaller pieces.

The rest of the code currently accesses the worker through method calls on the task, so this allowed them to keep doing that, but with one layer of abstraction in between.  In the future, I intend to look at making this more functional and declarative, which might, for example, look something like this.

```
const jobPath = './job'

const getRunningTask = src => task =>
   deepProp('childProcess.connected', task)
    ? task
    : new Task(require.resolve(src))

const getRunningJobTask = getRunningTask(jobPath)

const observeTask(task, event, f) => {
  task.on(event, f) // ewe, side effect :(
  return task
}
```